app/models/course_section.rb

```diff
@@ -37,6 +37,8 @@ class CourseSection < ActiveRecord::Base
   has_many :students, :through => :student_enrollments, :source => :user
   has_many :student_enrollments, :class_name => 'StudentEnrollment', :conditions => ['enrollments.workflow_state != ? AND enrollments.workflow_state != ? AND enrollments.workflow_state != ? AND enrollments.workflow_state != ?', 'deleted', 'completed', 'rejected', 'inactive'], :include => :user
   has_many :all_student_enrollments, :class_name => 'StudentEnrollment', :conditions => ['enrollments.workflow_state != ?', 'deleted'], :include => :user
+  has_many :tas, :through => :ta_enrollments, :source => :user
+  has_many :ta_enrollments, :class_name => 'Enrollment', :conditions => ['enrollments.workflow_state != ? AND enrollments.workflow_state != ? AND enrollments.workflow_state != ? AND enrollments.workflow_state != ? AND enrollments.type = \'TaEnrollment\'', 'deleted', 'completed', 'rejected', 'inactive'], :include => :user
```

Comment:
Can you double check these conditions?  It looks like you have the same one like 3 times.